clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/StoreRepushCommand.java

```diff
@@ -0,0 +1,279 @@
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * StoreRepushCommand contains the details of executing/processing repush command.
+ * We expect the command to comply with the following contract:
+ *
+ * Input:
+ *    <COMMAND> --store <store_name> --fabric <source_fabric> --password <credentials> [<EXTRA_COMMAND_ARGS>]
+ * Output:
+ *    success: link_to_running_task
+ *    failure: failure_reason
+ */
+
+public class StoreRepushCommand {
+  private static final Logger LOGGER = LogManager.getLogger(StoreRepushCommand.class);
+
+  // Store name.
+  private String store;
+  private Params params;
+  private Result result;
+  private List<String> shellCmd;
+
+  public StoreRepushCommand() {
+  }
+
+  public StoreRepushCommand(String store, Params params) {
+    this.store = store;
+    this.params = params;
+    this.shellCmd = generateShellCmd();
+  }
+
+  public void setParams(Params params) {
+    this.params = params;
+  }
+
+  public String getStore() {
+    return store;
+  }
+
+  public Result getResult() {
+    return result;
+  }
+
+  private List<String> generateRepushCommand() {
+    List<String> cmd = new ArrayList<>();
+    cmd.add(this.params.command);
+    cmd.add(String.format("--store '%s'", store));
+    cmd.add(String.format("--fabric '%s'", this.params.sourceFabric));
+    cmd.add(String.format("--password '%s'", this.params.password));
+    cmd.add(this.params.extraCommandArgs);
+    return cmd;
+  }
+
+  private List<String> generateShellCmd() {
+    List<String> shellCmd = new ArrayList<>();
+    // Start a shell process so that it contains the right PATH variables.
+    shellCmd.add("sh");
+    shellCmd.add("-c");
+    shellCmd.add(String.join(" ", generateRepushCommand()));
+    return shellCmd;
+  }
+
+  public List<String> getShellCmd() {
+    if (shellCmd == null) {
+      shellCmd = generateShellCmd();
+    }
+    return shellCmd;
+  }
+
+  public void processOutput(String output, int exitCode) {
+    result = new Result();
+    result.setStdOut(output);
+    result.setExitCode(exitCode);
+    result.parseStandardOutput();
+  }
+
+  public void execute() {
+    ProcessBuilder pb = new ProcessBuilder(getShellCmd());
+    // so we can ignore the error stream.
+    pb.redirectErrorStream(true);
+    int exitCode = -1;
+    BufferedReader reader = null;
+    String stdOut = StringUtils.EMPTY;
+    try {
+      Process process = pb.start();
+      reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+      String line;
+      StringBuilder buf = new StringBuilder();
+      while ((line = reader.readLine()) != null) {
+        buf.append(line);
+        buf.append('\n');
+      }
+      stdOut = buf.toString();
+      // remove trailing white spaces and new lines.
+      stdOut = stdOut.trim();
+
+      exitCode = process.waitFor();
+      processOutput(stdOut, exitCode);
+    } catch (IOException e) {
+      LOGGER.error("Error in executing command: {}", this, e);
```

Comment:
Shall we pop out the exception instead of just logging the error?
A bigger question:
How would the operator handle the repush failure for some of stores?

I was thinking we should make data recovery idempotent, for example, before doing a repush, data recovery tool should check whether there is an ongoing repush/data recovery or not, if there is one, data recovery tool should skip it and log a message and if the operator wants to execute data recovery regardless, the operator needs to terminate the ongoing repush.
So when the operator encounters an error, the operator can always re-run the same admin command with a full list of stores to re-run the leftover.
Does this make sense?

Reply:
Your thinking is still right and makes prefect sense. We should handle all that you mentioned. This change does not cover all that and it only builds the framework and implements a simple version of the executor that can trigger repush for a group of stores and reports results. We want to improve it in a follow up change to make it idempotent, and it mgiht require some changes on the controller api as well. Will create a ticket to track this effort.

Reply:
Sounds good, thanks